src/cripts/README.md

```diff
@@ -0,0 +1,453 @@
+    ...
+}
+
+// TBD more stuff here ?
+```
+
+## 3.4. Processing and modifying URLs
+
+Similarly to headers, URLs are important in Cripts, in all its various forms.
+Currently, Cript supports the following URLs:
+
+| URL name         | Description                                                    |
+| ---------------- | -------------------------------------------------------------- |
+| Client::Pristine | The pristine client request URL, which is immutable            |
+| Client::URL      | The clients request URL, which can be modified                 |
+| Cache::URL       | This is a special URL, used internally of ATS as the cache key |
+
+Getting these URLs follows a similar getter pattern to the headers:
+```
+borrow pristine = Pristine::URL::get();
+borrow client   = Client::URL::get();
+borrow cache    = Cache::URL::get();
+```
+
+Within a URL object, you can access all its various components via names following
+standard URL naming.
+
+| Identifier | Description                             |
+| ---------- | --------------------------------------- |
+| host       | The URL host                            |
+| port       | The URL port number                     |
+| path       | The URL path                            |
+| query      | The URL query parameters (all of them!) |
+
+Using this is easy (of course):
+```
+auto url = Client::URL;
+
+if (url.host == "www.example.com") {
+    ...
+}
+```
+
+The `path` components does support indexes, e.g. the first URL path
+component ("directory") can be read or written with `path[0]`.
+
+The query parameter has a set of additional features, which are particularly
+useful for modifying the cache key URLs. This includes sorting the list of
+query parameters, or adding and removing a query parameter by name. This is
+best explained with a real example:
+
+```
+borrow client_req = Client::Request::get();
+borrow c_url      = Cache::URL::get();
+
+c_url.query.sort(); // Sorts all query parameter by name first
+c_url.query["foo"].erase(); // Removes the foo query parameter, if it exists
+c_url.query["bar"] = "fie"; // Adds, or modifies, the bar query param to be "fie"
+
+if (client_req["X-Miles"]) {
+    c_url.path += "key_add=";
+    c_url.path += client_req["X-Miles"]; // Adds this header to the cache key
+}
+```
+
+With this in mind, a URL query path of
+
+```
+?foo=fum&z=xxx&d=yyy
+```
+
+gets normalized and modified to
+
+```
+?bar=fie&d=yyy&z=xxx
+```
+
+In addition, the cache key URL path is appended with an additional string
+extracted from the request headers.
+
+### 3.4.1. Special case: Cache Key URL
+
+## 3.5. Accessing and modifying connections
+
+There are essentially two possible connections being involved with a
+transaction: The client connections, and the origin server connection. The
+latter will not exist on a cache miss, and should therefore only be used
+in callbacks involving origin transactions (*cache misses*).
+
+Accessing the client connections is easily done with
```

Comment:
End with a `:`

Reply:
The README goes away once we land this and document, I'm not going to edit it.